src/callbacks_step/analysis_dg2d.jl

```diff
@@ -113,6 +113,21 @@ function analyze(::Val{:l2_divb}, du::AbstractArray{<:Any,4}, u, t,
   end |> sqrt
 end
 
+function analyze(::Val{:l2_divb}, du::AbstractArray{<:Any,4}, u, t,
```

Comment:
I wonder... this looks like the exact same like the version for regular MHD... And there are multiple functions like this with the *exact* same implementation, e.g., also for time step size calculations. Should we maybe add another abstract type which allows us to capture this?

However, I don't know if this makes sense or even works, since it would have to be a type that is both indicative of the "ideal glm mhd" concept and of the "2D" concept. @ranocha What's your take on this? Figure this out or leave it as is until we have more such cases?

Reply:
Actually it is "not" the exact same implementation, since in the version I pushed here the indices have to be shifted to the left. I fix this now.

Reply:
fixed in ed324b5

Reply:
If we had exactly the same code twice, I would strongly prefer coding it only once, e.g. using dispatch on a union. If the code is sufficiently different, that's of course another story...